test/e2e/provisioning/all_e2e_test.go

```diff
@@ -0,0 +1,86 @@
+// +build e2e
+
+package provisioning
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+const (
+	do_manifest  = "./testdata/machine-digitalocean.yaml"
+	aws_manifest = "./testdata/machine-aws.yaml"
+	hz_manifest  = "./testdata/machine-hetzner.yaml"
+)
+
+// TestDigitalOceanProvisioning - a test suite that exercises digital ocean provider
+// by requesting nodes with different combination of container runtime type, container runtime version and the OS flavour.
+//
+// note that tests require  a valid API Token that is read from the DO_E2E_TEST_TOKEN environmental variable.
+func TestDigitalOceanProvisioningE2E(t *testing.T) {
+	t.Parallel()
+
+	// test data
+	doToken := os.Getenv("DO_E2E_TESTS_TOKEN")
+	if len(doToken) == 0 {
+		t.Fatal("unable to run the test suite, DO_E2E_TESTS_TOKEN environement varialbe cannot be empty")
+	}
+
+	// act
+	params := fmt.Sprintf("<< DIGITALOCEAN_TOKEN >>=%s", doToken)
+	runScenarios(t, scenarios, params, do_manifest, "do")
+}
+
+// TestAWSProvisioning - a test suite that exercises AWS provider
+// by requesting nodes with different combination of container runtime type, container runtime version and the OS flavour.
+func TestAWSProvisioningE2E(t *testing.T) {
+	t.Parallel()
+
+	// test data
+	awsKeyID := os.Getenv("AWS_E2E_TESTS_KEY_ID")
+	awsSecret := os.Getenv("AWS_E2E_TESTS_SECRET")
+	if len(awsKeyID) == 0 || len(awsSecret) == 0 {
+		t.Fatal("unable to run the test suite, AWS_E2E_TESTS_KEY_ID or AWS_E2E_TESTS_SECRET environment variables cannot be empty")
+	}
+
+	// act
+	params := fmt.Sprintf("<< AWS_ACCESS_KEY_ID >>=%s,<< AWS_SECRET_ACCESS_KEY >>=%s", awsKeyID, awsSecret)
+	runScenarios(t, scenarios, params, aws_manifest, "aws")
+}
+
+var hzScenarios = []scenario{
+	scenarios[0],
+	scenarios[1],
+	scenarios[2],
+
+	{
+		name:                    "scenario 4 CentOS Docker 1.12",
+		osName:                  "centos",
+		containerRuntime:        "docker",
+		containerRuntimeVersion: "1.12",
```

Comment:
version `17.03` not supported.

Reply:
That particular test case tests `centos`. I'm not sure if the others (`DO`, `AWS` etc.) support this flavour. Just by looking at the source code, I'm not sure if we support `docker 1.12` for CoreOS or Ubuntu.

Reply:
here is the reference: https://github.com/kubermatic/machine-controller/issues/129

Reply:
Yeah `centos` works for everything except vsphre (allthought its just untested there, so it could potentially work).

Refs:
* https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/aws/provider.go#L110
* https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/digitalocean/provider.go#L87
* https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/hetzner/provider.go#L50